encrypted-config-value/src/main/java/com/palantir/config/crypto/LegacyEncryptedValue.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.config.crypto;
+
+
+import com.palantir.config.crypto.algorithm.aes.AesEncryptedValue;
+import com.palantir.config.crypto.algorithm.aes.AesKey;
+import com.palantir.config.crypto.algorithm.aes.ImmutableAesEncryptedValue;
+import com.palantir.config.crypto.algorithm.rsa.ImmutableRsaEncryptedValue;
+import com.palantir.config.crypto.algorithm.rsa.RsaEncryptedValue;
+import com.palantir.config.crypto.algorithm.rsa.RsaOaepEncrypter;
+import com.palantir.config.crypto.algorithm.rsa.RsaPrivateKey;
+import java.util.Arrays;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class LegacyEncryptedValue extends EncryptedValue {
+
+    @Value.Parameter
+    public abstract byte[] getCiphertext();
+
+    @Override
+    public final String decrypt(KeyWithType kwa) {
+        EncryptedValue translatedValue;
```

Comment:
When a legacy value is decrypted, we match legacy behavior and interpret the ciphertext based on the provided key.

The `xValueFromLegacy` functions instantiate an `EncryptedValue` of the new form in a manner that matches the specific parameters that were used for legacy encryption.